packages/material-ui/src/Typography/Typography.d.ts

```diff
@@ -46,14 +50,14 @@ export interface TypographyTypeMap<P = {}, D extends React.ElementType = 'span'>
     /**
      * Applies the theme typography styles.
      */
-    variant?: Variant | 'inherit';
+    variant?: OverridableStringUnion<TypographyVariantDefaults, TypographyPropsVariantOverrides>;
     /**
      * The component maps the variant prop to a range of different HTML element types.
      * For instance, subtitle1 to `<h6>`.
      * If you wish to change that mapping, you can provide your own.
      * Alternatively, you can use the `component` prop.
      */
-    variantMapping?: Partial<Record<Variant, string>>;
+    variantMapping?: Partial<Record<OverridableStringUnion<TypographyVariantDefaults, TypographyPropsVariantOverrides>, string>>;
```

Comment:
How do we want to reflect this in the `propTypes`, so it still would make at least some form of validation?